mock/mock.go

```diff
@@ -281,15 +281,17 @@ func (m *Mock) findExpectedCall(method string, arguments ...interface{}) (int, *
 
 func (m *Mock) findClosestCall(method string, arguments ...interface{}) (*Call, string) {
 	var diffCount int
+	var repeatability int
 	var closestCall *Call
 	var err string
 
 	for _, call := range m.expectedCalls() {
 		if call.Method == method {
 
 			errInfo, tempDiffCount := call.Arguments.Diff(arguments)
-			if tempDiffCount < diffCount || diffCount == 0 {
+			if tempDiffCount < diffCount || diffCount == 0 || (tempDiffCount == diffCount && call.Repeatability > repeatability) {
```

Comment:
Would it make sense to extract this logic into a separate function? It seems to be getting relatively complex and we definitely need to add unit tests for this new logic to make sure there are no regressions regarding previous behaviour.

Reply:
I was going for a minimal change. I have now added a more extensive rewrite of the method along with more testing.